vignettes/summary_selecting.Rmd

```diff
@@ -0,0 +1,229 @@
+---
+title: "A quick summary of selecting possibilities in `datawizard`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{A quick summary of selecting possibilities in `datawizard`}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup, echo=FALSE, message=FALSE}
+library(datawizard)
+library(poorman)
+library(htmltools)
+
+set.seed(123)
+iris <- iris[sample(nrow(iris), 10), ]
+row.names(iris) <- NULL
+
+row <- function(...) {
+  div(
+    class = "custom_note",
+    ...
+  )
+}
+```
+
+```{css, echo=FALSE}
+.custom_note {
+  border-left: solid 5px hsl(220, 100%, 30%);
+  background-color: hsl(220, 100%, 95%);
+  padding: 5px;
+  margin-bottom: 10px
+}
+```
+
+# Selecting variables
+
+## Quoted names
+
+This is the most simple way to select one or several variables. Just use a character
+vector containing variables names, like in base R.
+
+```{r}
+get_columns(iris, c("Sepal.Length", "Petal.Width"))
+```
+
+## Unquoted names
+
+It is also possible to use unquoted names. This is useful if we use the `tidyverse`
+and want to be consistent about the way variable names are passed.
+
+```{r}
+iris %>%
+  group_by(Species) %>%
+  standardise(Petal.Length) %>%
+  ungroup()
+```
+
+
+## Formulas
+
+You can also use formulas to select variables:
+
+```{r}
+get_columns(iris, ~ Sepal.Length + Petal.Width)
+```
+
+This was mostly useful to use selection more easily in custom functions before
+`datawizard` 0.6.0, and is kept available for backward compatibility. 
+
+
+
+## Positions
+
+In addition to variable names, `select` can also take a numeric vector that indicates
+the position of the variables to select in the dataframe.
+
+```{r}
+get_columns(iris, c(1, 2, 5))
+```
+
+
+## Functions
+
+We can pass a function to `select`. This function will be applied to all columns
+and has to return `TRUE` or `FALSE`. For example, if we want to keep only numeric
+columns, we can use `is.numeric`.
+
+```{r}
+get_columns(iris, is.numeric)
+```
+
+Note that we can also define our own functions and pass them in `select`, 
+*as long as they return `TRUE` or `FALSE`* when applied to a column.
+
+```{r}
+my_function <- function(i) {
+  is.numeric(i) && mean(i, na.rm = TRUE) > 3.5
+}
+
+get_columns(iris, my_function)
+```
+
+
+## Patterns
+
+Most of the time, our data will be quite bigger than `iris`. It would be tedious
+to write the names of variables we want to select, and it would be fragile to rely
+on variable positions in the dataset as they may change later. In this case, we
+can use four select helpers: `starts_with()`, `ends_with()`, `contains()`, and
+`regex()`. The former three can take several patterns, while the latter takes
+a single regular expression.
```

Comment:
```suggestion
With larger datasets, it would be tedious
to write the names of variables to select, and it would be fragile to rely
on variable positions as they may change later. To this end, we
can use four select helpers: `starts_with()`, `ends_with()`, `contains()`, and
`regex()`. The first three can take several patterns, while `regex()` takes
a single regular expression.
```